fast64_internal/oot/exporter/export.py

```diff
@@ -0,0 +1,219 @@
+    sceneC.sceneCollisionC = ootCollisionToC(outScene.collision)
+    sceneC.sceneCutscenesC = convertCutsceneToC(outScene)
+
+    for room in outScene.rooms.values():
+        name = room.roomName()
+        sceneC.roomMainC[name] = convertRoomLayers(room)
+        sceneC.roomMeshInfoC[name] = convertRoomShapeData(room.mesh)
+        sceneC.roomMeshC[name] = convertRoomModel(room, textureExportSettings)
+
+    return sceneC
+
+
+def ootSceneIncludes(outScene: OOTScene):
+    sceneIncludeData = CData()
+    includeFiles = [
+        "ultra64.h",
+        "z64.h",
+        "macros.h",
+        f"{outScene.sceneName()}.h",
+        "segment_symbols.h",
+        "command_macros_base.h",
+        "variables.h",
+    ]
+
+    sceneIncludeData.source = "\n".join([f'#include "{fileName}"' for fileName in includeFiles]) + "\n\n"
+    return sceneIncludeData
+
+
+def ootCombineSceneFiles(sceneC: OOTSceneC):
+    # only used for single file export
+    newSceneC = CData()
+    newSceneC.append(sceneC.sceneMainC)
+
+    if sceneC.sceneTexturesIsUsed():
+        newSceneC.append(sceneC.sceneTexturesC)
+
+    newSceneC.append(sceneC.sceneCollisionC)
+
+    if sceneC.sceneCutscenesIsUsed():
+        for i in range(len(sceneC.sceneCutscenesC)):
+            newSceneC.append(sceneC.sceneCutscenesC[i])
+
+    return newSceneC
+
+
+def ootCreateSceneHeader(sceneC: OOTSceneC):
+    # writes the scene.h file
+    sceneHeader = CData()
+    sceneHeader.append(sceneC.sceneMainC)
+
+    if sceneC.sceneTexturesIsUsed():
+        sceneHeader.append(sceneC.sceneTexturesC)
+
+    sceneHeader.append(sceneC.sceneCollisionC)
+
+    if sceneC.sceneCutscenesIsUsed():
+        for sceneCs in sceneC.sceneCutscenesC:
+            sceneHeader.append(sceneCs)
+
+    for roomMainC in sceneC.roomMainC.values():
+        sceneHeader.append(roomMainC)
+
+    for roomMeshInfoC in sceneC.roomMeshInfoC.values():
+        sceneHeader.append(roomMeshInfoC)
+
+    for roomMeshC in sceneC.roomMeshC.values():
+        sceneHeader.append(roomMeshC)
+
+    return sceneHeader
+
+
+def ootPreprendSceneIncludes(outScene: OOTScene, fileData: CData):
+    exportFile = ootSceneIncludes(outScene)
+    exportFile.append(fileData)
+    return exportFile
+
+
+def exportScene(
+    inSceneObj: Object,
+    transformMatrix: Matrix,
+    f3dType: str,
+    isHWv1: bool,  # is hardware v1
+    sceneName: str,
+    dlFormat: DLFormat,
+    savePNG: bool,
+    exportInfo: ExportInfo,
+    bootToSceneOptions: OOTBootupSceneOptions,
+):
+    # circular import fix
+    from .scene_folder import modifySceneFiles
+    from .spec import modifySegmentDefinition
+    from ..oot_level_writer import ootConvertScene
+
+    checkObjectReference(inSceneObj, "Scene object")
+    isCustomExport = exportInfo.isCustomExportPath
+    exportPath = exportInfo.exportPath
+
+    outScene: OOTScene = ootConvertScene(
+        inSceneObj, transformMatrix, f3dType, isHWv1, sceneName, dlFormat, not savePNG
+    )
```

Comment:
I'm not a fan of this circular import fix. What we could do instead is move this block of code into `oot_level_writer.py` right before calling `exportScene`, then have `exportScene` take in an `OOTScene` argument. This way `export.py` doesn't have any dependency on `oot_level_writer.py`. This would also remove a lot of the parameters for `exportScene`.